modules/hal_bouffalolab/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2021-2025 Gerson Fernando Budke <nandojve@gmail.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_SOC_FAMILY_BFLB)
+  zephyr_library_named(hal_bouffalolab)
+
+  # This is necessary to proper select architecture in the SDK
+  zephyr_compile_definitions(
+    ARCH_RISCV
+  )
```

Comment:
@tejlmand for comment

Reply:
I removed all the entries of ARCH_RISCV in the hal and this is not necessary anymore.